### DISCUSSION TITLE
Making Modals be "Active" in React components

### DISCUSSION THREAD

**Discussion by hawwokitty** [React95/React95#458] [Q&A] [answered]
Hello! I'm using React95 for a project in react but i cant seem to get the modals to be "active" when clicking them. I have them separated into different components, so they just render in the order which the component is placed, and none of them are shown as active :( is there a fix to this?

link to my WIP project: https://github.com/hawwokitty/Portfolio

**Comment by ggdaltoso** [comment, marked as answer]
I can see you're wrapping each component with a `ModalProvider`:
 - [StreamingComp](https://github.com/hawwokitty/Portfolio/blob/main/my-portfolio/src/components/StreamingComp.jsx#L13)
 - [PaintComp](https://github.com/hawwokitty/Portfolio/blob/main/my-portfolio/src/components/PaintComp.jsx#L14)

Instead of doing this, please make sure you're wrapping your application just once. A suggestion would be in [`main.jsx`](https://github.com/hawwokitty/Portfolio/blob/main/my-portfolio/src/main.jsx) file:

```jsx
import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
import { ModalProvider } from '@react95/core'

import App from './App.jsx'
import './index.css'
import './App.css'
import '@react95/core/GlobalStyle';
import '@react95/core/themes/candy.css';

createRoot(document.getElementById('root')).render(
  <ModalProvider >
    <App />
  </ModalProvider >,
)
```

Let me know if it worked 👍

**Reply by hawwokitty** [reply]
that fixed it!! thank you so much 😄
